src/main/java/org/bahmni/reports/template/TSIntegrationDiagnosisService.java

```diff
@@ -0,0 +1,120 @@
+        return config.getPatientAttributes() != null ? config.getPatientAttributes() : new ArrayList<>();
+    }
+
+    static List<String> getPatientAddresses(TSIntegrationDiagnosisLineReportConfig config) {
+        return config.getPatientAddresses() != null ? config.getPatientAddresses() : new ArrayList<>();
+    }
+
+
+    default void loadTempTable(Connection connection, String tempTableName, String parentCode, Properties tsProperties, String descendantsUrlTemplate, HttpClient httpClient) {
+        int offset = 0;
+        int pageSize = getDefaultPageSize(tsProperties);
+        try {
+            String createSqlStmt = MessageFormat.format(CREATE_SQL_TEMPLATE, tempTableName);
+            String insertSqlStmt = MessageFormat.format(INSERT_SQL_TEMPLATE, tempTableName);
+            Statement statement = connection.createStatement();
+            statement.execute(createSqlStmt);
+            try (PreparedStatement pstmtInsert = connection.prepareStatement(insertSqlStmt)) {
+
+                TSPageObject pageObject = null;
+                do {
+                    try {
+                        pageObject = fetchDescendantsByPagination(parentCode, pageSize, offset, "en", httpClient, descendantsUrlTemplate);
+                    } catch (IOException e) {
+                        throw new RuntimeException();
+                    }
+                    List<String> codes = pageObject.getCodes();
+                    for (int batchcount = 0; batchcount < codes.size(); batchcount++) {
+                        pstmtInsert.setString(1, codes.get(batchcount));
+                        pstmtInsert.addBatch();
+                    }
+                    pstmtInsert.executeBatch();
+                    offset += pageSize;
+                } while (offset < pageObject.getTotal());
+            }
+        } catch (SQLException e) {
+            logger.error("Error occured while making database call to " + tempTableName + " table");
+            throw new RuntimeException();
+        }
+    }
+
+    default TSPageObject fetchDescendantsByPagination(String terminologyCode, int pageSize, int offset, String localeLanguage, HttpClient httpClient, String descendantsUrlTemplate) throws IOException {
+        String url = MessageFormat.format(descendantsUrlTemplate, terminologyCode, pageSize, offset, localeLanguage);
+        String responseStr = httpClient.get(URI.create(url));
+        ObjectMapper mapper = new ObjectMapper();
+        try {
+            return mapper.readValue(responseStr, TSPageObject.class);
+        } catch (JsonProcessingException e) {
+            throw new RuntimeException();
+        }
+    }
+
+    default int getDefaultPageSize(Properties tsProperties) {
+        String pageSize = System.getenv("REPORTS_TS_PAGE_SIZE");
+        if (pageSize == null) pageSize = tsProperties.getProperty("ts.defaultPageSize");
+        if (pageSize != null) {
```

Comment:
What happens if pageSize is null? It can be null if ts.defaultPageSize is not set.

Reply:
Then why do you have additional if (pageSize !=null) check on line 90. That isn't needed.